package_control/commands/advanced_install_package_command.py

```diff
@@ -91,15 +92,13 @@ def run(self):
         # Allow packages to properly disable
         time.sleep(0.7)
 
+        def do_reenable_package(package_name):
+            type_ = 'install' if package_name not in self.installed else 'upgrade'
+            self.reenable_package(package_name, type_)
+
         for package in self.packages:
-            self.manager.install_package(package)
-
-            # We use a wrapper function since this call is in a loop, so
-            # directly using the "package" variable would cause the value to
-            # change between when we set the callback and executed it.
-            def reenable_package(package_name):
-                def do_reenable_package():
-                    type_ = 'install' if package_name not in self.installed else 'upgrade'
-                    self.reenable_package(package_name, type_)
-                return do_reenable_package
-            sublime.set_timeout(reenable_package(package), 700)
+            result = self.manager.install_package(package)
+
+            # Do not reenable if installation deferred until next restart
+            if result is not None:
+                sublime.set_timeout(functools.partial(do_reenable_package, package), 700)
```

Comment:
Similar to above, there should be some explanation as to why a partial is being created here.